sympy/combinatorics/fp_groups.py

```diff
@@ -1343,13 +1381,17 @@ def reidemeister_presentation(fp_grp, H, C=None):
     if not C:
         C = coset_enumeration_r(fp_grp, H)
     C.compress(); C.standardize()
-    define_schreier_generators(C)
+    define_schreier_generators(C, rel=rel)
     reidemeister_relators(C)
     gens, rels = C._schreier_generators, C._reidemeister_relators
+    # To-do change scherier generators correspondance to newset of gens.s
     gens, rels = simplify_presentation(gens, rels, change_gens=True)
 
     C.schreier_generators = tuple(gens)
     C.reidemeister_relators = tuple(rels)
+    if rel:
```

Comment:
As I said somewhere above, it would make more sense to return a homomorphism rather than a list of generators. And in any case, `rel` looks to much like short for `relator` which is confusing. This can be `homomorphism` if you return one.